modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/resource_factory_impl.ftl

```diff
@@ -164,6 +166,10 @@ public class ${schemaName}ResourceFactoryImpl implements ${schemaName}Resource.F
 
 		${schemaVarName}Resource.setContextCompany(company);
 
+		if ((httpServletRequest != null) && (httpServletRequest.getAttribute(WebKeys.CTX) == null)) {
+			httpServletRequest.setAttribute(WebKeys.CTX, ServletContextPool.get(StringPool.BLANK));
+		}
+
```

Comment:
@4lejandrito then added it in the factory right before setter invocation. Looks a bit clearer to me but if you don't like it I'm fine with reverting this back to the baseResourceImpl setter